### PR TITLE
[fuse-perf] Add fuse-benchmark tool

### LINKED ISSUE #652: fuse-benchmark  tool
**Summary**

```
❯ echo "select sum(number) from numbers(100)" | fuse-benchmark -i 10 -c 2
Loaded 1 queries.

Queries executed: 10.

localhost:9000, queries 10, QPS: 1017.760, RPS: 101775.991, MiB/s: 0.776, result RPS: 1017.760, result MiB/s: 0.008.

0.000%		0.001 sec.
10.000%		0.002 sec.
20.000%		0.002 sec.
30.000%		0.002 sec.
40.000%		0.002 sec.
50.000%		0.002 sec.
60.000%		0.002 sec.
70.000%		0.002 sec.
80.000%		0.002 sec.
90.000%		0.002 sec.
95.000%		0.003 sec.
99.000%		0.003 sec.
99.900%		0.003 sec.
99.990%		0.003 sec.

```

### PR DESCRIPTION
**Pull request by sundy-li**
## Summary

Add fuse-benchmark tool 

## Changelog

- New Feature

## Related Issues

Fixes #652 

## Test Plan

Performance tests

### CODE REVIEW

File: fusequery/query/src/servers/clickhouse/clickhouse_stream.rs [hunk 1]
```diff
@@ -12,24 +12,31 @@ use common_arrow::arrow::array::*;
...
+            input,
+            block_index: 0,
+            schema
+        }
     }
 
     pub fn convert_block(&self, block: DataBlock) -> Result<ClickHouseBlock> {
         let mut result = ClickHouseBlock::new();
-        if block.is_empty() {
+        if block.is_empty() && block.num_columns() == 0 {
```
Comment (BohuTANG) [inline fusequery/query/src/servers/clickhouse/clickhouse_stream.rs:39]: is_empty() includes rows and columns zero check:
https://github.com/datafuselabs/datafuse/blob/0136e6452dc0245ae33f7264a0852f277f13bc53/common/datablocks/src/data_block.rs#L54-L72

**Comment by sundy-li** [comment]
Example:

```shell
echo "select avg(number) from numbers(10000000)" |  ./target/debug/fuse-benchmark -c 3  -i 10
    Finished dev [unoptimized + debuginfo] target(s) in 0.16s

Queries 10, QPS: 3.238341968911917, RPS: 32383419.68911917, MiB/s: 30.883235634917423.

0.000%          0.151 sec.
10.000%         0.151 sec.
20.000%         0.162 sec.
30.000%         0.212 sec.
40.000%         0.271 sec.
50.000%         0.272 sec.
60.000%         0.301 sec.
70.000%         0.301 sec.
80.000%         0.316 sec.
90.000%         0.419 sec.
95.000%         0.683 sec.
99.000%         0.683 sec.
99.900%         0.683 sec.
99.990%         0.683 sec.

```

File: fusequery/query/src/bin/fuse-benchmark.rs [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+                }
+                bench.executed.fetch_add(1, Ordering::Relaxed);
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn report_text(bench: BenchmarkRef) -> Result<()> {
+    eprintln!();
```
Comment (zhang2014) [inline fusequery/query/src/bin/fuse-benchmark.rs:230]: maybe we should remove it?

File: fusequery/query/src/interpreters/interpreter.rs [hunk 3]
```diff
@@ -2,12 +2,17 @@
...
 
+use common_datavalues::DataSchemaRef;
+use common_datavalues::DataSchemaRefExt;
 use common_exception::Result;
 use common_streams::SendableDataBlockStream;
 
 #[async_trait::async_trait]
 pub trait IInterpreter: Sync + Send {
     fn name(&self) -> &str;
+    fn schema(&self) -> DataSchemaRef {
```
Comment (zhang2014) [inline fusequery/query/src/interpreters/interpreter.rs:13]: style. For functions with default implement, maybe we should add new line?

**Review by zhang2014 — APPROVED** [review]
LGTM

File: fusequery/query/src/servers/clickhouse/clickhouse_stream.rs [hunk 4]
```diff
@@ -115,6 +122,14 @@ impl Stream for ClickHouseStream {
     type Item = Result<ClickHouseBlock>;
 
     fn poll_next(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        // Some drivers will skip the first block for it recognizes the first block as schema
+        // So we make the first block to be an empty block
+        if self.block_index == 0 {
+            self.block_index += 1;
+            let block = DataBlock::empty_with_schema(self.schema.clone());
+            return Poll::Ready(Some(self.convert_block(block)));
+        }
```
Comment (tisonkun) [inline fusequery/query/src/servers/clickhouse/clickhouse_stream.rs:131]: This block seems the root cause of https://github.com/datafuselabs/datafuse/issues/703.

What is its background? We don't always return schema as the title, at least in explain situation.

Comment (sundy-li) [reply fusequery/query/src/servers/clickhouse/clickhouse_stream.rs:131]: Oh, it's just to be compatible with `clickhouse-rs`,  https://github.com/suharev7/clickhouse-rs/blob/async-await/src/types/query_result/stream_blocks.rs#L85

I'll take a look at it.

**Pull request merged**